[IEEE ICMLA 2019](https://www.icmla-conference.org/icmla19/)

[The Data Science landscape: foundations, tools, and practical applications](https://www.icmla-conference.org/icmla19/links/tutorialAM.htm)

# 3. Using data to answer questions

In this part, we will look at the correlation between female literacy and fertility (defined as the average number of children born per woman) throughout the world. For ease of analysis and interpretation, we will work with the illiteracy rate.

This dataset comes from $\color{red}{\text{TODO: ADD SOURCE HERE}}$.

Our objective is to test this hypothesis:

> There is a positive correlation between illiteracy and fertility, i.e. the higher the illiteracy rate, the higher the average number of children born per woman is

## Loading and inspecting the dataset

In [1]:
import pandas as pd

df = pd.read_csv('data/female_literacy_fertility.csv')

In [2]:
df.shape

(162, 5)

In [3]:
df.columns

Index(['Country', 'Continent', 'female literacy', 'fertility', 'population'], dtype='object')

In [4]:
df.head()

Country Continent  female literacy  fertility     population
0      Chine       ASI             90.5      1.769  1,324,655,000
1       Inde       ASI             50.8      2.682  1,139,964,932
2        USA       NAM             99.0      2.077    304,060,000
3  Indonésie       ASI             88.8      2.132    227,345,082
4     Brésil       LAT             90.2      1.827    191,971,506

In [5]:
df.describe()

female literacy   fertility
count       162.000000  162.000000
mean         80.107407    2.878673
std          23.052415    1.427597
min          12.600000    0.966000
25%          66.425000    1.823250
50%          90.000000    2.367500
75%          98.500000    3.880250
max         100.000000    7.069000

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 162 entries, 0 to 161
Data columns (total 5 columns):
Country            162 non-null object
Continent          162 non-null object
female literacy    162 non-null float64
fertility          162 non-null float64
population         162 non-null object
dtypes: float64(2), object(3)
memory usage: 6.5+ KB


By inspecting the dataset, we found out that:
    
1. It has 162 entries (`shape`).
1. Each entry seems to represent one country (`head()`). We will check this assumption soon.
1. Each entry has the female literacy rate and average fertility (children born per woman) for that country, plus other fields we are not interested in (`head()` again).
1. There some low literacy rates (`describe()`). We will check this too soon.
1. The fertility rates seem reasonable (also from `describe()`).
1. There are no missing values in any row or column (`info()`).
1. The representation of each column (number or string) seems correct (also from `info()`).

Let's check the assumption we made in item 2, that there is one entry per country.

In [7]:
print(df.Country.nunique())

162


There are 162 unique countries names, matching the number of rows in the dataset. Therefore, each row is a unique country.

Now let's check the countries with suspiciously low literacy rate.

In [8]:
display(df[df['female literacy'] <= 30])

Country Continent  female literacy  fertility  population
15       Ethiopie        AF             22.8      5.211  80,713,434
39    Afghanistan       ASI             12.6      6.505  29,021,099
60   Burkina Faso        AF             21.6      5.841  15,233,884
62          Niger        AF             15.1      7.069  14,704,318
66           Mali        AF             18.2      5.405  12,705,736
72          Tchad        AF             21.9      6.081  10,913,667
79         Guinée        AF             26.4      5.329   9,833,055
85          Bénin        AF             28.1      5.378   8,662,086
101  Sierra Leone        AF             28.9      5.165   5,559,853

These countries are the ones we would expect to have low literacy. The values also match the [World Bank's female literacy dataset, around 2010](https://data.worldbank.org/indicator/SE.ADT.LITR.FE.ZS?end=2010&start=1970).

At this point we understand and trust our data. We can move on to testing the hypothesis.

## Preparing the data

This the hypothesis we want to test (copied from the first section):

> There is a positive correlation between illiteracy and fertility, i.e. the higher the illiteracy rate, the higher the average number of children born per woman is

We do not need the complete dataset to test the hypothesis. We just need the literacy rate and the fertility number. We will split them out of the dataset.

Since we the hypothesis is phrased in terms of _illiteracy_ and the dataset has _literacy_, we will also change the literacy rate to illiteracy rate. This makes writing the code a bit more natural (we don't have to think in terms of a reverse relationship all the time), preventing possible bugs (good data representation reduces [extraneous cognitive load](https://en.wikipedia.org/wiki/Cognitive_load)).

Also, once present the results to the stakeholders, or even review in details how we got there, the graphs and code will match more closely how they phrase (and frame) the hypothesis.

In [9]:
fertility = df['fertility']
illiteracy = 100 - df['female literacy']

In [10]:
print(fertility.head())

0    1.769
1    2.682
2    2.077
3    2.132
4    1.827
Name: fertility, dtype: float64


In [11]:
print(illiteracy.head())

0     9.5
1    49.2
2     1.0
3    11.2
4     9.8
Name: female literacy, dtype: float64
